activerecord/lib/active_record/migration/compatibility.rb

```diff
@@ -44,11 +73,18 @@ def change(name, type, **options)
             super
           end
 
+          def index(column_name, **options)
+            options[:name] = legacy_index_name(name, column_name) if options[:name].nil?
+            indexes << [column_name, options]
```

Comment:
Can this call `super` instead? It shouldn't make a difference, but that just keeps the "real" implementation all in the same place.

```suggestion
            indexes << [column_name, options]
```